### PR TITLE
fix!: replace htmllabels and htmllabelitem with annotation based labels

### LINKED ISSUE #1446: Remove HTMLLabels and HTMLLabelItem
It was deprecated in HC7 and won't work https://api.highcharts.com/highcharts/labels
Create API for annotations instead.

### PR DESCRIPTION
**Pull request by ugur-vaadin**
## Description

The counterparts for `HTMLLabels` and `HTMLLabelItem`  in Highcharts were deprecated in favour of annotation based labels. This PR introduces the integration for Annotations. While annotations in Highcharts contain more than just the labels, this PR only implements the necessary parts to retain our current behaviour.

This is a breaking change.

Related web component [PR](https://github.com/vaadin/web-components/pull/5221)
For the tests to pass, the web component PR has to be merged first.

Partially fixes #1446 

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

### CODE REVIEW

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/Configuration.java [hunk 1]
```diff
@@ -1163,4 +1145,34 @@ public void addColorAxis(ColorAxis axis) {
...
+     * @return Labels
+     */
+    public List<AnnotationItemLabel> getLabels() {
+        return getLabelsAnnotation().getLabels();
+    }
+
+    /**
+     * Sets labels that can be positioned anywhere in the chart area
+     */
+    public void setLabels(AnnotationItemLabel... labels) {
```
Comment (DiegoCardoso) [inline vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/Configuration.java:1159]: Javadoc for params missing.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/Configuration.java [hunk 2]
```diff
@@ -1163,4 +1145,34 @@ public void addColorAxis(ColorAxis axis) {
...
+        getLabelsAnnotation().setLabels(labels);
+    }
+
+    /**
+     * Adds a single label
+     *
+     * @see #setLabels(AnnotationItemLabel...)
+     */
+    public void addLabel(AnnotationItemLabel label) {
```
Comment (DiegoCardoso) [inline vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/Configuration.java:1168]: Javadoc for the parameter missing.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+     */
+    public AnnotationItemLabelMockPointOptions getPoint() {
+        return point;
+    }
+
+    /**
+     * Sets the {@link AnnotationItemLabelMockPointOptions} that contains the
+     * coordinate data for the label
+     */
+    public void setPoint(AnnotationItemLabelMockPointOptions point) {
```
Comment (DiegoCardoso) [inline vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java:36]: Javadoc for params missing.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+     * @see #setStyle(Style)
+     */
+    public Style getStyle() {
+        return style;
+    }
+
+    /**
+     * Label style options
+     */
+    public void setStyle(Style style) {
```
Comment (DiegoCardoso) [inline vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java:50]: Javadoc for params missing.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+     * @see #setText(String)
+     */
+    public String getText() {
+        return text;
+    }
+
+    /**
+     * Value to be displayed
+     */
+    public void setText(String text) {
```
Comment (DiegoCardoso) [inline vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java:64]: Javadoc for params missing.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+     * @see #setUseHTML(Boolean)
+     */
+    public Boolean getUseHTML() {
+        return useHTML;
+    }
+
+    /**
+     * Whether to enable HTML parsing for the label contents
+     */
+    public void setUseHTML(Boolean useHTML) {
```
Comment (DiegoCardoso) [inline vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java:78]: Javadoc for params missing.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabelMockPointOptions.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+     * @see #setX(Number)
+     */
+    public Number getX() {
+        return x;
+    }
+
+    /**
+     * Sets the horizontal offset of the label within chart
+     */
+    public void setX(Number x) {
```
Comment (DiegoCardoso) [inline vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabelMockPointOptions.java:36]: Javadoc for params missing.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabelMockPointOptions.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+     * @see #setY(Number)
+     */
+    public Number getY() {
+        return y;
+    }
+
+    /**
+     * Sets the vertical offset of the label within chart
+     */
+    public void setY(Number y) {
```
Comment (DiegoCardoso) [inline vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabelMockPointOptions.java:50]: Javadoc for params missing.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabelMockPointOptions.java [hunk 9]
```diff
@@ -0,0 +1,53 @@
+package com.vaadin.flow.component.charts.model;
+
+/**
+ * Contains coordinates for {@link AnnotationItemLabel}
+ */
+public class AnnotationItemLabelMockPointOptions
```
Comment (DiegoCardoso) [inline vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabelMockPointOptions.java:6]: I know the name of the class is related to the HC types, but I am worried about this one being a bit long.

**Review by DiegoCardoso — CHANGES_REQUESTED** [review]

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabelMockPointOptions.java (hunk 9 above)
Comment (ugur-vaadin) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabelMockPointOptions.java]: Renamed it to `AnnotationItemLabelPoint`.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabelMockPointOptions.java (hunk 8 above)
Comment (ugur-vaadin) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabelMockPointOptions.java]: Addressed in the latest commit.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabelMockPointOptions.java (hunk 7 above)
Comment (ugur-vaadin) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabelMockPointOptions.java]: Addressed in the latest commit.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java (hunk 6 above)
Comment (ugur-vaadin) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java:90]: Addressed in the latest commit.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java (hunk 5 above)
Comment (ugur-vaadin) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java:73]: Addressed in the latest commit.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java (hunk 4 above)
Comment (ugur-vaadin) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java:56]: Addressed in the latest commit.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java (hunk 3 above)
Comment (ugur-vaadin) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/AnnotationItemLabel.java]: Addressed in the latest commit.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/Configuration.java (hunk 2 above)
Comment (ugur-vaadin) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/Configuration.java:1173]: Addressed in the latest commit.

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/Configuration.java (hunk 1 above)
Comment (ugur-vaadin) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/Configuration.java:1162]: Addressed in the latest commit.

**Review by DiegoCardoso — APPROVED** [review]
LGTM

**Pull request merged**